examples/example_app/example_app/views.py

```diff
@@ -77,6 +79,72 @@ class ExampleModelUIViewSet(views.NautobotUIViewSet):
     queryset = ExampleModel.objects.all()
     serializer_class = serializers.ExampleModelSerializer
     table_class = tables.ExampleModelTable
+    object_detail_content = ui.ObjectDetailContent(
+        panels=(
+            ui.ObjectFieldsPanel(
+                section=ui.SectionChoices.LEFT_HALF,
+                weight=100,
+                fields="__all__",
+            ),
+            # A table of objects derived dynamically in `get_extra_context()`
+            ui.ObjectsTablePanel(
+                section=ui.SectionChoices.RIGHT_HALF,
+                weight=100,
+                context_table_key="dynamic_table",
+                max_display_count=3,
+            ),
+            # A table of non-object data with staticly defined columns
+            ui.DataTablePanel(
+                section=ui.SectionChoices.RIGHT_HALF,
+                label="Custom Table 1",
+                weight=200,
+                context_data_key="data_1",
+                columns=["col_1", "col_2", "col_3"],
+                column_headers=["Column 1", "Column 2", "Column 3"],
+            ),
+            # A table of non-object data with dynamic (render-time) columns
+            ui.DataTablePanel(
+                section=ui.SectionChoices.FULL_WIDTH,
+                label="Custom Table 2",
+                weight=100,
+                context_data_key="data_2",
+                context_columns_key="columns_2",
+                context_column_headers_key="column_headers_2",
+            ),
+        ),
+    )
+
+    def get_extra_context(self, request, instance):
+        context = super().get_extra_context(request, instance)
+        if self.action == "retrieve":
+            # Add dynamic table of objects for custom panel
+            context["dynamic_table"] = CircuitTable(Circuit.objects.restrict(request.user, "view"))
+            # Add non-object data for object detail view custom tables
+            context["data_1"] = [
+                {"col_1": "value_1a", "col_2": "value_2", "col_3": "value_3", "col_4": "not shown"},
+                {"col_1": "value_1b", "col_2": None},
+            ]
+            context["data_2"] = [
```

Comment:
Another good idea might be to move those table definitions to a separate file and then import as dict or import and call some function to generate the proper data. Then looking at `get_extra_context` would be much simpler with just `context["data_2"], context["columns_2"], context["column_headers_2"] = example_table_data()`.